dali/ft/filecopy.cpp

```diff
@@ -2422,6 +2434,40 @@ void FileSprayer::pullParts()
 }
 
 
+//Execute a parallel write to a remote part, but each slave writes the entire contents of the file
+void FileSprayer::pushWholeParts()
+{
+    bool needCalcCRC = calcCRC();
+    LOG(MCdebugInfoDetail, job, "Calculate CRC = %d", needCalcCRC);
+    //Create a slave for each of the target files, but execute it on the node corresponding to the first source file
+    ForEachItemIn(idx, targets)
+    {
+        TargetLocation & cur = targets.item(idx);
+        SocketEndpoint ep;
+        ForEachItemIn(idx3, partition)
+        {
+            PartitionPoint & cur = partition.item(idx3);
+            if (cur.whichOutput == idx)
```

Comment:
@ghalliday - I think I need an overview of how targets and partitions are setup when target for export is an azure URL..
Looking at the export code I thought it would end up mapping through to setTarget(IFileDescriptor...) where really that was a pseudo IFileDesc that contained a single RemoteFilename representing the azure URL..
Which would mean there's 1 target (in 'targets').

But I assume that # partitions = # of sources, in order to create a transfer thread for each.. so looks like I'm missing something.

Reply:
The despray command has a wrap=1 option which sets the number of target parts to be the same as the source.  Although this code should work for any N:M copy.  It ensures that each target file part is only written to from a single slave.

Reply:
I have also updated the despray code (in nother PR) so that if wrap is used, and there is no $P$ in the target filename it implicitly adds one to match the default.